### PR TITLE
Add unit test for CQ validation for unsupported configurations of TAS

### LINKED ISSUE #3516: Add unit test for CQ validation for unsupported configurations (the inactiveReason)
**What would you like to be added**:
Add unit test for CQ validation for unsupported configurations (the inactiveReason)

**Why is this needed**:
Increase test coverage.
Follow up to discussion: https://github.com/kubernetes-sigs/kueue/pull/3320#discussion_r1819422375 

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by mszadkow**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Test CQ validation for unsupported configurations of TAS.

#### Which issue(s) this PR fixes:

Fixes #3516 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/cache/clusterqueue_test.go [hunk 1]
```diff
@@ -1274,3 +1274,144 @@ func TestCohortLendable(t *testing.T) {
 		t.Errorf("Unexpected cohort lendable (-want,+got):\n%s", diff)
 	}
 }
+
+func TestClusterQueueReadinessWithTAS(t *testing.T) {
```
Comment (mimowo) [inline pkg/cache/clusterqueue_test.go:1278]: This looks great, we only use the exported functions in this test which makes it much more reliable than `TestClusterQueueUpdateWithAdmissionCheck`. Could you try to migrate the `TestClusterQueueUpdateWithAdmissionCheck` into this one? And rename `TestClusterQueueReadinessWithTAS` to just `TestClusterQueueReadiness`. `enableTAS` could be a paratemer

Comment (mszadkow) [reply pkg/cache/clusterqueue_test.go:1278]: `TestClusterQueueReadiness` already exists, so maybe we could merge this one there and simply make the `TestClusterQueueUpdateWithAdmissionCheck` to use exported functions.

File: pkg/cache/clusterqueue_test.go [hunk 2]
```diff
@@ -1274,3 +1274,144 @@ func TestCohortLendable(t *testing.T) {
...
+			cqCache.AddOrUpdateAdmissionCheck(mkAC)
+
+			acWithPR := utiltesting.MakeAdmissionCheck("pr-check").ControllerName(kueue.ProvisioningRequestControllerName).Active(metav1.ConditionTrue).Obj()
+			cqCache.AddOrUpdateAdmissionCheck(acWithPR)
+
+			if err := cqCache.AddClusterQueue(ctx, tc.cq); err != nil {
+				t.Fatalf("Inserting clusterQueue %s in cache: %v", tc.cq.Name, err)
+			}
+
+			if tc.updatedCq != nil {
```
[the remark this answers is not in the document]
Comment (mszadkow) [reply pkg/cache/clusterqueue_test.go:1397]: works

File: pkg/cache/clusterqueue_test.go (hunk 1 above)
Comment (mimowo) [reply pkg/cache/clusterqueue_test.go:1278]: I would just merge them under `TestClusterQueueReadiness` name. Look at the asserts there: https://github.com/kubernetes-sigs/kueue/pull/3541/files#diff-e11e73ca517b5546e51925b6e6299ff79841efda37d39243ccee269d731de277R597-R598. The intention of the other test seems to test what this test does.

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve
Please try to follow up with merging `TestClusterQueueReadinessWithTAS` and `TestClusterQueueUpdateWithAdmissionCheck` into `TestClusterQueueReadiness`

**Pull request merged**